packages/components/addon/components/hds/copy/button/index.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export const DEFAULT_SIZE = 'medium';
+export const SIZES = ['small', 'medium'];
+export const DEFAULT_ICON = 'clipboard-copy';
+export const SUCCESS_ICON = 'clipboard-checked';
+export const ERROR_ICON = 'clipboard';
+export const DEFAULT_STATUS = 'idle';
+
+export default class HdsCopyButtonComponent extends Component {
+  @tracked status = DEFAULT_STATUS;
+  @tracked timer;
+
+  get icon() {
```

Comment:
I feel like this code could be cleverer, but then maybe it wouldn't be so easy to read.

Reply:
I observe that the alert code is doing more than just the status/icon mapping and maybe it's appropriate to be more complex there; I don't mind changing it here, but after thinking about the mapping, I wondered if it was overkill since it was only default/success/error. WDYT?

(also, added comments)

Reply:
Works for me!